crates/anvil-polkadot/src/api_server/server.rs

```diff
@@ -990,15 +998,56 @@ impl ApiServer {
 
         let latest_block = self.latest_block();
 
-        let Some(ReviveAccountInfo { account_type: AccountType::Contract(contract_info), .. }) =
-            self.backend.read_revive_account_info(latest_block, address)?
-        else {
-            return Ok(());
+        let account_id = self.get_account_id(latest_block, address)?;
+
+        let maybe_system_account_info =
+            self.backend.read_system_account_info(latest_block, account_id.clone())?;
+        let nonce = maybe_system_account_info.as_ref().map(|info| info.nonce).unwrap_or_default();
+
+        if maybe_system_account_info.is_none() {
+            self.set_frame_system_balance(
+                latest_block,
+                account_id.clone(),
+                substrate_runtime::currency::DOLLARS,
+            )?;
+        }
+
+        let trie_id = match self.backend.read_revive_account_info(latest_block, address)? {
```

Comment:
nit: Add a comment specifying that we create the contract account if it does not exist already.